Solutions/STM32F4DISCOVERY/TinyBooter/TinyBooter.proj

```diff
@@ -23,6 +23,8 @@
         <ExtraTargets>$(ExtraTargets);CompressBin</ExtraTargets>
         <ScatterFileDefinition>scatterfile_bootloader_$(COMPILER_TOOL).$(SCATTER_EXT)</ScatterFileDefinition>
         <EXEScatterFileDefinition>scatterfile_bootloader_$(COMPILER_TOOL).$(SCATTER_EXT)</EXEScatterFileDefinition>
+        <!-- newlib-nano does not support long-long and long-double formatting, use standard newlib instead -->

```

Comment:
Another option would be to change the CLR code so it doesn't use printf style formatting from the CRT for long and double and instead does it's own thing. Thus, allowing for newlib-nano in all cases. It would be ideal if we could figure out how to eliminate the duplication of the CRT completely (e.g. Tinybooter has a copy of the CRT and so does TinyCLR ) do we really NEED two copies of the CRT in the device?)